geoh5py/groups/property_group_table.py

```diff
@@ -0,0 +1,162 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from abc import ABC
+from typing import TYPE_CHECKING
+from uuid import UUID
+
+import numpy as np
+
+from ..data import DataAssociationEnum
+from ..data.primitive_type_enum import DataTypeEnum
+
+
+if TYPE_CHECKING:  # pragma: no cover
+    from .property_group import PropertyGroup
+
+
+class PropertyGroupTable(ABC):
+    """
+    A class to store the information of a property group.
+
+    :param property_group: The property group to extract the data from.
+    """
+
+    locations_keys = ("X", "Y", "Z")
+
+    def __init__(self, property_group: PropertyGroup):
+        if not hasattr(property_group, "property_group_type"):
+            raise TypeError("'property_group' must be a PropertyGroup object.")
+
+        if hasattr(property_group.parent, "collar"):
+            raise NotImplementedError(
+                "PropertyGroupTable is not supported for Drillhole objects."
+            )
```

Comment:
We should be able to support `drillholes` since `drillholes.locations` returns a 2D array of desurveyd xyz positions.

Reply:
Also check for assiociation type in `(DataAssociationEnum.CELL, DataAssociationEnum.VERTEX, DataAssociationEnum.DEPTH)`

Reply:
That's too complicated.
The data of drillholes are not associated with the locations but with the depth or the from-to. 
and a drillhole can contain a bunch of them.

Reply:
Yea you are right.